neural_lam/vis.py

```diff
@@ -128,8 +145,12 @@ def plot_spatial_error(error, obs_mask, data_config, title=None, vrange=None):
     else:
         vmin, vmax = vrange
 
+    extent = data_config.get_xy_extent("state")
+
     # Set up masking of border region
-    mask_reshaped = obs_mask.reshape(*data_config.grid_shape_state)
+    mask_reshaped = obs_mask.reshape(
+        list(data_config.grid_shape_state.values.values())
+    )
```

Comment:
1. Here (and in all of `plot_spatial_error`)  `data_config` is still used. I believe this should be changed to use the datastore.
2. This seems very inconvenient and hard to understand, for getting the grid shape (the `.values.values()`). This would work right?
```suggestion
    mask_reshaped = obs_mask.reshape(
        data_config.grid_shape_state.y,
        data_config.grid_shape_state.x,
    )
```
(this also applies to the similar line below)

Reply:
I have fixed this in https://github.com/leifdenby/neural-lam/commit/7e46194b91bd2b7c00850bbb925b59e1fa98d091 and for getting `mask_reshaped` I used the unstacking method of the datastore: https://github.com/leifdenby/neural-lam/commit/7e46194b91bd2b7c00850bbb925b59e1fa98d091#diff-a7d63bf900568d9f8688b77f43fd291bc6d87720d7a23542d68d831f2466d326R151